pkg/cmd/provisioning/aws/delete.go

```diff
@@ -76,7 +78,100 @@ func deleteOIDCBucket(client aws.Client, bucketName, namePrefix string) error {
+			}
+
+			_, err = client.DeleteCloudFrontOriginAccessIdentity(&cloudfront.DeleteCloudFrontOriginAccessIdentityInput{
+				Id:      originAccessIdentity.Id,
+				IfMatch: getCloudFrontOriginAccessIdentityOutput.ETag,
+			})
+			if err != nil {
+				return errors.Wrapf(err, "failed to delete the CloudFront origin access identity with ID %s", *originAccessIdentity.Id)
+			}
+			log.Printf("CloudFront origin access identity with ID  %s deleted", *originAccessIdentity.Id)
+		}
+	}
+	return nil
+}
+
+// deleteCloudFrontDistribution deletes the CloudFront distribution if created
+func deleteCloudFrontDistribution(client aws.Client, namePrefix string) error {
+	ListCloudFrontDistributionsOutput, err := client.ListCloudFrontDistributions(&cloudfront.ListDistributionsInput{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to fetch a list of CloudFront distributions")
+	}
+	for _, distribution := range ListCloudFrontDistributionsOutput.DistributionList.Items {
+		listTagsForCloudFrontResourceOutput, err := client.ListTagsForCloudFrontResource(&cloudfront.ListTagsForResourceInput{
+			Resource: distribution.ARN,
+		})
+		if err != nil {
+			return errors.Wrapf(err, "failed to fetch tags for CloudFront distribution with ID %s", *distribution.Id)
+		}
+
+		for _, tag := range listTagsForCloudFrontResourceOutput.Tags.Items {
+			if *tag.Key == fmt.Sprintf("%s/%s", ccoctlAWSResourceTagKeyPrefix, namePrefix) {
+				getCloudFrontDistributionOutput, err := client.GetCloudFrontDistribution(&cloudfront.GetDistributionInput{
+					Id: distribution.Id,
+				})
+				if err != nil {
+					return errors.Wrapf(err, "failed to get CloudFront Distribution with ID %v", distribution.Id)
+				}
+
+				getCloudFrontDistributionOutput.Distribution.DistributionConfig.Enabled = awssdk.Bool(false)
+
+				updateCloudFrontDistributionOutput, err := client.UpdateCloudFrontDistribution(&cloudfront.UpdateDistributionInput{
+					Id:                 distribution.Id,
+					IfMatch:            getCloudFrontDistributionOutput.ETag,
+					DistributionConfig: getCloudFrontDistributionOutput.Distribution.DistributionConfig,
+				})
+				if err != nil {
+					return errors.Wrapf(err, "failed to disable CloudFront Distribution with ID %v", distribution.Id)
+				}
+
+				for {
+					getCloudFrontDistributionOutput, err := client.GetCloudFrontDistribution(&cloudfront.GetDistributionInput{
+						Id: distribution.Id,
+					})
+					if err != nil {
+						return errors.Wrapf(err, "failed to get CloudFront Distribution with ID %s", *distribution.Id)
+					}
+
+					if *getCloudFrontDistributionOutput.Distribution.Status == cloudFrontDistributionDeployedStatus {
+						log.Printf("CloudFront distribution with ID %s is successfully disabled", *distribution.Id)
+						break
+					}
+					log.Printf("Waiting for CloudFront distribution with ID %s to be disabled...", *distribution.Id)
```

Comment:
Nit: add duration to these waiting messages.

```suggestion
					log.Printf("Waiting %s for CloudFront distribution with ID %s to be disabled...", cloudFrontDistributionStatusCheckDelay, *distribution.Id)
```